haigha/connection.py

```diff
@@ -404,7 +404,10 @@ def read_frames(self):
         # Send a heartbeat (if needed)
         self._channels[0].send_heartbeat()
 
-        data = self._transport.read(self._heartbeat)
+        # Wait for 2 heartbeat intervals before giving up. See AMQP 4.2.7:
+        # "If a peer detects no incoming traffic (i.e. received octets) for two heartbeat intervals or longer,
+        # it should close the connection"
+        data = self._transport.read(self._heartbeat*2)
```

Comment:
I've tested this today with the latest RabbitMQ 3.5.3.
I've initialized a `Connection` with `heartbeat=60`. As a result:
1. When we negotiate connection parameters with RabbitMQ, we send this parameter. RabbitMQ then sends a heartbeat frame *every 60 seconds*.
2. Haigha timeouts after 60 seconds with no arriving bytes - which happens all the time. It does so silently, so nothing happens (which is a point I will address in future pull requests), but still - it's wrong. According to the spec, we should wait 120 seconds in this case.